### PR TITLE
Fixes stacks incorrectly stacking with subtypes

### LINKED ISSUE #76157: Durathread and Cotton share bundles
## Reproduction:
1. Place a partial stack of raw cotton on a table
2. Place a partial stack of durathread on the same table
3. The cotton stack will merge into the durathread stack
Couldn't fully test because the server crashed shortly after and rebooted

### PR DESCRIPTION
**Pull request by Momo8289**
## About The Pull Request
Makes the type check to determine if two items will stack more strict, so that subtypes that shouldn't stack don't. Fixed #76157
## Why It's Good For The Game
Fixes a bug, prevents losing items from incorrect stacking
## Changelog
:cl:
fix: Items that shouldn't have been stacking should no longer stack.
/:cl:

### CODE REVIEW

**Comment by Sealed101** [comment]
please change `Addresses` to `Fixed` in your PR body to link the issue to the PR properly

**Review by MrMelbert — CHANGES_REQUESTED** [review]
This will prevent mapped in amount subtypes (`/obj/item/stack/sheet/mineral/plasma/five`) from merging directly with its correct type (`/obj/item/stack/sheet/mineral/plasma`)

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -566,7 +566,8 @@
  * - [inhand][boolean]: Whether or not the stack to check should act like it's in a mob's hand.
  */
 /obj/item/stack/proc/can_merge(obj/item/stack/check, inhand = FALSE)
-	if(!istype(check, merge_type))
+	// We don't use istype here, since that will match subtypes, and stack things that shouldn't stack
+	if(check.type != merge_type)
```
Comment (MrMelbert) [inline code/game/objects/items/stacks/stack.dm:570]: Maybe it should be...?
```suggestion
	if(!istype(check) || check.merge_type != merge_type)
```

Comment (Momo8289) [reply code/game/objects/items/stacks/stack.dm:570]: Tested it, this seems to work well

**Comment by Momo8289** [comment]
> edit: thanks, for future reference [these](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) are all the keywords that can be used to link issues/PRs
Awesome, thanks!

**Comment by ChungusGamer666** [comment]
istype check should be as it was before

```dm
if(!istype(check, merge_type) || check.merge_type != merge_type)
```

**Pull request merged**